hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultPortableReader.java

```diff
@@ -20,44 +20,49 @@
 import com.hazelcast.nio.BufferObjectDataInput;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.serialization.ClassDefinition;
-import com.hazelcast.nio.serialization.FieldDefinition;
 import com.hazelcast.nio.serialization.FieldType;
 import com.hazelcast.nio.serialization.HazelcastSerializationException;
 import com.hazelcast.nio.serialization.Portable;
 import com.hazelcast.nio.serialization.PortableReader;
+import com.hazelcast.query.extractor.ValueCallback;
+import com.hazelcast.query.extractor.ValueCollector;
+import com.hazelcast.query.extractor.ValueReader;
+import com.hazelcast.query.extractor.ValueReadingException;
+import com.hazelcast.query.impl.getters.MultiResult;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Set;
-import java.util.regex.Pattern;
 
-public class DefaultPortableReader implements PortableReader {
+import static com.hazelcast.internal.serialization.impl.PortableUtils.getPortableArrayCellPosition;
 
-    private static final Pattern NESTED_FIELD_PATTERN = Pattern.compile("\\.");
+/**
+ * Can't be accessed concurrently
+ */
+public class DefaultPortableReader extends ValueReader implements PortableReader {
```

Comment:
If you need to query n objects, how many of these DefaultPortableReader instances are created?